MultiSource/Benchmarks/Olden/mst/makegraph.c

```diff
@@ -41,21 +41,21 @@ static int hashfunc(unsigned int key)
 static void AddEdges(int count1, Graph retval, int numproc, 
                      int perproc, int numvert, int j) 
 {
-  Vertex tmp;
-  Vertex helper[MAXPROC];
+  Vertex tmp = NULL;
+  UncheckedVertex helper checked[MAXPROC];
   int i;
 
   for (i=0; i<numproc; i++) {
-    helper[i] = retval->vlist[i];
+    helper[i] = (UncheckedVertex)retval->vlist[i];
   }
 
-  for (tmp = retval->vlist[j]; tmp; tmp=tmp->next) 
+  for (tmp = retval->vlist[j]; tmp; tmp=tmp->next)
     {
       for (i=0; i<numproc*perproc; i++) 
         {
           int pn,offset,dist;
-          Vertex dest;
-          Hash hash;
+          UncheckedVertex dest;
```

Comment:
`checked-c-convert` suggests this should become `Vertex dest;`. This actually totally fails to compile, because of new rules around casting to `ptr<T>`, The exact error is:

```
/Users/ashe2/Code/checkedc-llvm-test-suite/MultiSource/Benchmarks/Olden/mst/makegraph.c:65:22: error: expression has no bounds, right-hand side of assignment expected to have bounds because the left-hand side has bounds
              dest = ((helper[pn])+offset);
                     ^~~~~~~~~~~~~~~~~~~~~
/Users/ashe2/Code/checkedc-llvm-test-suite/MultiSource/Benchmarks/Olden/mst/makegraph.c:65:22: error: expression has no bounds, cast to ptr<T> expects source to have bounds
              dest = ((helper[pn])+offset);
                     ^~~~~~~~~~~~~~~~~~~~~
```

I'm going to file an issue against these error messages, because I'm not sure the restrictions are entirely well thought out.